config/diff.go

```diff
@@ -12,6 +12,14 @@ import (
 // ErrUnsupportedType is returned if the type is not implemented
 var ErrUnsupportedType = errors.New("unsupported type")
 
+// includeNilValuesForKeys prevents removing certain keys with null values.
+// This is useful when default value for a given field is not nil, user set this field for null in oldConfig,
+// and we want to keep this field as null in the newConfig.
+var includeNilValuesForKeys = map[string]struct{}{
```

Comment:
This looks not correct to me. Why is this only a problem for ephemeral-storage and not for cpu and memory? Also I don't think this is the right way to do it, since it seems this is more a logic problem that values that are nil before are now not there anymore, which should be solved through a generic solution